x/capability/keeper/keeper.go

```diff
@@ -332,15 +356,14 @@ func (sk ScopedKeeper) GetCapability(ctx sdk.Context, name string) (*types.Capab
 		// to still have the capability in the go map since changes to
 		// go map do not automatically get reverted on tx failure,
 		// so we delete here to remove unnecessary values in map
-		delete(sk.capMap, index)
+		// TODO: Delete index correctly from capMap by storing some reverse lookup
+		// in-memory map. Issue: https://github.com/cosmos/cosmos-sdk/issues/7805
 		return nil, false
 	}
 
 	cap := sk.capMap[index]
 	if cap == nil {
-		// delete key from store to remove unnecessary mapping
```

Comment:
Was there a reason this was expected to happen before?

Reply:
As far as we can tell, no. Maybe some prior bug which has since been fixed.

Reply:
No, afaik this should never happen. I think it used to just ignore the inconsistency and move on, now we panic